### PR TITLE
Endpoint validate if Loomie still exists

### PR DESCRIPTION
**Pull request by andres123dbh**
Try PR:
image
Loomie exists:
image
Loomie not exists:
image

### CODE REVIEW

File: api/routes/router.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ func SetupRoutes(engine *gin.Engine) {
 
 	// Loomies
 	engine.POST("/loomies/near", middlewares.MustProvideAccessToken(), controllers.HandleNearLoomies)
+	engine.GET("/loomies/exists", middlewares.MustProvideAccessToken(), controllers.HandleValidateLoomieExists)
```
Comment (PedroChaparro) [inline api/routes/router.go:29]: Usually `GET` routes doesn't receive any payload / JSON content, so, this endpoint should be `POST`.

File: api/controllers/loomies.go [hunk 2]
```diff
@@ -167,3 +168,33 @@ func HandleNearLoomies(c *gin.Context) {
...
+
+	if loomie_id.LoomieId == "" {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": true, "message": "No Loomie ID"})
+		return
+	}
+
+	exists, _ := models.ValidateLoomieExists(loomie_id.LoomieId)
+
+	if !exists {
+		c.AbortWithStatusJSON(http.StatusNotFound, gin.H{
```
Comment (PedroChaparro) [inline api/controllers/loomies.go:189]: Please, follow this schema for the responses on ALL the endpoints: 

```json
{
  "error": "boolean:", 
  "message": "string"
}
```

You can add other fields if needed, but always include the `error` and `message` fields.

File: api/models/loomies.go [hunk 3]
```diff
@@ -169,3 +169,23 @@ func GetNearWildLoomies(coordinates interfaces.Coordinates) ([]interfaces.WildLo
 
 	return loomies, nil
 }
+
+func ValidateLoomieExists(loomie_id string) (bool, error) {
```
Comment (PedroChaparro) [inline api/models/loomies.go:173]: You can just return the error, if there is an error you can infer that the Loomie doesn't exists.

**Review by PedroChaparro — CHANGES_REQUESTED** [review]
Check the requested changes and also, document your changes in the `.github/docs/openapi.yml` file, you can use the online swagger editor an then, copy the content of the file to out local file.

File: .github/docs/openapi.yml [hunk 4]
```diff
@@ -538,6 +538,61 @@ paths:
...
+                type: object
+                properties:
+                  error:
+                      type: boolean
+                      example: false
+                  message: 
+                      type: string
+                      example: "Loomie exists"
+        "400":
+          description: Bad request. Mayby the loomie id field are empty.
```
Comment (PedroChaparro) [inline .github/docs/openapi.yml:572]: Mayby -> Maybe
Are empty -> Is empty

File: .github/docs/openapi.yml [hunk 5]
```diff
@@ -538,6 +538,61 @@ paths:
...
+              schema:
+                $ref: "#/components/schemas/FailResponse"
+        "401":
+          description: The access token wasn't provided or isn't valid.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/FailResponse"
+        "404":
+          description: Loomie not exists in collection wild_loomies
```
Comment (PedroChaparro) [inline .github/docs/openapi.yml:584]: Not exists -> Doesn't exists
Collection wild_loomies -> wild_loomies collection.

File: .github/docs/openapi.yml [hunk 6]
```diff
@@ -538,6 +538,61 @@ paths:
...
+            application/json:
+              schema:
+                type: object
+                properties:
+                  error:
+                      type: boolean
+                      example: true
+                  message: 
+                      type: string
+                      example: "Loomie no exists"
```
Comment (PedroChaparro) [inline .github/docs/openapi.yml:595]: No exists -> Doesn't exists.

File: api/controllers/loomies.go [hunk 7]
```diff
@@ -167,3 +168,33 @@ func HandleNearLoomies(c *gin.Context) {
...
+	var loomie_id interfaces.LoomieExistsForm
+
+	if err := c.BindJSON(&loomie_id); err != nil {
+		fmt.Println(err)
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": true, "message": "Bad request"})
+		return
+	}
+
+	if loomie_id.LoomieId == "" {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": true, "message": "No Loomie ID"})
```
Comment (PedroChaparro) [inline api/controllers/loomies.go:182]: I suggest changing the message to "No Loomie ID was provided".

File: api/controllers/loomies.go [hunk 8]
```diff
@@ -167,3 +168,33 @@ func HandleNearLoomies(c *gin.Context) {
...
+		return
+	}
+
+	err := models.ValidateLoomieExists(loomie_id.LoomieId)
+
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusNotFound, gin.H{
+			"error":   true,
+			"message": "Loomie no exists",
```
Comment (PedroChaparro) [inline api/controllers/loomies.go:191]: No exists -> Doesn't exists.

**Review by PedroChaparro — CHANGES_REQUESTED** [review]

File: api/routes/router.go [hunk 9]
```diff
@@ -26,6 +26,7 @@ func SetupRoutes(engine *gin.Engine) {
 
 	// Loomies
 	engine.POST("/loomies/near", middlewares.MustProvideAccessToken(), controllers.HandleNearLoomies)
+	engine.POST("/loomies/exists", middlewares.MustProvideAccessToken(), controllers.HandleValidateLoomieExists)
```
Comment (PedroChaparro) [inline api/routes/router.go:29]: What about receiving the Loomie Id as an endpoint parameter. Eg. `/loomies/exists/:id`. At this way the endpoint can use the `GET` method and can work without sending any JSON payload. Let me know if you'll implement this.

Comment (PedroChaparro) [reply api/routes/router.go:29]: Also consider what @Woynert said (Returning the Loomie ID).

**Review by PedroChaparro — APPROVED** [review]

**Review by SilviaPabon — APPROVED** [review]

**Pull request merged**